contracts/account/utils/draft-ERC7579Utils.sol

```diff
@@ -0,0 +1,289 @@
+        }
+        callData = executionCalldata[20:];
+    }
+
+    /**
+     * @dev Encodes a batch of executions. See {decodeBatch}.
+     */
+    function encodeBatch(Execution[] memory executionBatch) internal pure returns (bytes memory executionCalldata) {
+        return abi.encode(executionBatch);
+    }
+
+    /**
+     * @dev Decodes a batch of executions. See {encodeBatch}.
+     */
+    function decodeBatch(bytes calldata executionCalldata) internal pure returns (Execution[] calldata executionBatch) {
+        assembly ("memory-safe") {
+            let ptr := add(executionCalldata.offset, calldataload(executionCalldata.offset))
+            // Extract the ERC7579 Executions
+            executionBatch.offset := add(ptr, 32)
+            executionBatch.length := calldataload(ptr)
+        }
+    }
+
+    /**
+     * @dev Executes a `call` to the target with the provided {ExecType}.
+     */
+    function _call(
+        uint256 index,
+        ExecType execType,
+        address target,
+        uint256 value,
+        bytes calldata data
+    ) private returns (bytes memory) {
+        (bool success, bytes memory returndata) = target.call{value: value}(data);
+        return _validateExecutionMode(index, execType, success, returndata);
+    }
+
+    /**
+     * @dev Validates the execution mode and returns the returndata.
+     */
+    function _validateExecutionMode(
+        uint256 index,
+        ExecType execType,
+        bool success,
+        bytes memory returndata
+    ) private returns (bytes memory) {
+        if (execType == ERC7579Utils.EXECTYPE_DEFAULT) return Address.verifyCallResult(success, returndata);
+        if (execType == ERC7579Utils.EXECTYPE_TRY) {
+            if (!success) emit ERC7579TryExecuteFail(index, returndata);
+            return returndata;
+        }
+        revert ERC7579UnsupportedExecType(execType);
+    }
+}
+
+// Operators
+using {_eqCallTypeGlobal as ==} for CallType global;
+using {_eqExecTypeGlobal as ==} for ExecType global;
+using {_eqModeSelectorGlobal as ==} for ModeSelector global;
+using {_eqModePayloadGlobal as ==} for ModePayload global;
+
+/// @dev Compares two `CallType` values for equality.
+function _eqCallTypeGlobal(CallType a, CallType b) pure returns (bool) {
+    return CallType.unwrap(a) == CallType.unwrap(b);
+}
+
+/// @dev Compares two `ExecType` values for equality.
+function _eqExecTypeGlobal(ExecType a, ExecType b) pure returns (bool) {
+    return ExecType.unwrap(a) == ExecType.unwrap(b);
+}
+
+/// @dev Compares two `ModeSelector` values for equality.
+function _eqModeSelectorGlobal(ModeSelector a, ModeSelector b) pure returns (bool) {
+    return ModeSelector.unwrap(a) == ModeSelector.unwrap(b);
+}
+
+/// @dev Compares two `ModePayload` values for equality.
+function _eqModePayloadGlobal(ModePayload a, ModePayload b) pure returns (bool) {
+    return ModePayload.unwrap(a) == ModePayload.unwrap(b);
+}
```

Comment:
For the record and as discussed, the intention was to signal the function shouldn't be used directly since we recommend the operator. That said, I'll be fine with the current naming, otherwise I'd remove the `_` and `Global` suffix.